blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala

```diff
@@ -27,54 +27,56 @@ object BlazeClient {
       onShutdown: F[Unit])(implicit F: Sync[F]): Client[F] =
     Client(
       Kleisli { req =>
-        val key = RequestKey.fromRequest(req)
-        val submitTime = Instant.now()
+        F.suspend {
```

Comment:
They have different type signatures:
```scala
def suspend[A](thunk: => F[A]): F[A]
def delay[A](thunk: => A): F[A]
```
and suspend fit the bill since we're also generating an `F[_]` result and just needed to capture even more side effects.